drivers/accel/adxl355/adxl355.c

```diff
@@ -0,0 +1,652 @@
+*******************************************************************************/
+int32_t adxl355_get_status_register(struct adxl355_dev *dev, uint8_t *reg_value)
+{
+	int32_t status;
+
+	status = adxl355_get_register_value(dev, ADXL355_STATUS, reg_value);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Reads the number of FIFO entries register value.
+ *
+ * @param dev       - The device structure.
+ * @param reg_value - Register value.
+ *
+ * @return status   - Result of the reading procedure.
+*******************************************************************************/
+int32_t adxl355_get_nb_of_fifo_entries(struct adxl355_dev *dev, uint8_t *reg_value)
+{
+	int32_t status;
+
+	status = adxl355_get_register_value(dev, ADXL355_FIFO_ENTRIES, reg_value);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Sets the number of FIFO entries register value.
+ *
+ * @param dev       - The device structure.
+ * @param reg_value - Register value.
+ *
+ * @return status   - Result of the writing procedure.
+*******************************************************************************/
+int32_t adxl355_set_fifo_samples(struct adxl355_dev *dev, uint8_t reg_value)
+{
+	int32_t status;
+
+	status = adxl355_set_register_value(dev, ADXL355_FIFO_SAMPLES, reg_value);
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Sets the measurement range register value.
+ *
+ * @param dev       - The device structure.
+ * @param reg_value - Register value.
+ *
+ * @return status   - Result of the writing procedure.
+*******************************************************************************/
+int32_t adxl355_set_range(struct adxl355_dev *dev, adxl355_range range_val)
+{
+	int32_t status;
+
+	status = adxl355_set_register_value(dev, ADXL355_RANGE, range_val);
+	dev->range = range_val;
+
+	return status;
+}
+
+/***************************************************************************//**
+ * @brief Writes the low-pass filter settings.
+ *
+ * @param dev         - The device structure.
+ * @param odr_lpf_val - Low-pass filter settings.
+ *
+ * @return status     - Result of the writing procedure.
+*******************************************************************************/
+int32_t adxl355_set_odr_lpf(struct adxl355_dev *dev, adxl355_odr_lpf odr_lpf_val)
+{
+	int32_t status = 0;
+	uint8_t reg_value;
+	adxl355_op_mode current_op_mode;
+
+	// The lpf settings cannot be changed when the device is in measurement mode.
+	// The lpf settings can be changed only when the device's operation mode is
+	// Standby.
+
+	current_op_mode = dev->op_mode;
+
+	if (current_op_mode != ADXL355_STANDBY)
+	{
+		status = adxl355_set_op_mode(dev, ADXL355_STANDBY);
+	}
+
+	status |= adxl355_get_register_value(dev, ADXL355_FILTER, &reg_value);
```

Comment:
You should not `status |= function()` but simply `status = function()` and return immediately on error.